File: LondonDataServices.IDecide.Portal.Client/src/models/patients/patientCodeRequest.ts
export class PatientCodeRequest {
    public nhsNumber: string;
    public verificationCode: string;
    public notificationPreference: string;
    public generateNewCode: boolean;

    constructor(patientCode: PatientCodeRequest) {
        this.nhsNumber = patientCode.nhsNumber || "";
        this.verificationCode = patientCode.verificationCode || "";
        this.notificationPreference = patientCode.notificationPreference || "";
        this.generateNewCode = patientCode.generateNewCode || false;
    }
}

File: LondonDataServices.IDecide.Manage.Server/Models/PatientCodeRequest.cs
// ---------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------

namespace LondonDataServices.IDecide.Manage.Server.Models
{
    public class PatientCodeRequest
    {
        public string NhsNumber { get; set; }
        public string VerificationCode { get; set; }
        public string NotificationPreference { get; set; }
        public bool GenerateNewCode { get; set; } = false;
    }
}
